mesos/cli/cmds/frameworks.py

```diff
@@ -0,0 +1,144 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import, print_function
+
+import blessings
+import prettytable
+
+from .. import cli
+from ..master import CURRENT as MASTER
+
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
+
+
+parser = cli.parser(
+    description="frameworks list"
+)
+
+parser.add_argument(
+    "-t", "--task-breakdown", action="store_true",
+    help="show task status breakdown"
+)
+
+parser.add_argument(
+    "-i", "--inactive", action="store_true",
+    help="show inactive frameworks as well"
+)
+
+
+class TaskStats:
+    '''
+    To avoid going through the task state for every task state, this class acts as
+    a cache that generates the stats for the task in one call and returns it if it
+    has been computed already
+    '''
+    def __init__(self):
+        self.stats_cache = dict()
+
+    def get(self, framework_id, tasks, state):
+        # check if we have things in the cache already
+        if framework_id in self.stats_cache.keys():
+            return self._lookup(self.stats_cache[framework_id], state)
+
+        state_count = dict()
+        for task in tasks:
+            task_state = task['state']
+            if task_state in state_count.keys():
+                state_count[task_state] += 1
+            else:
+                state_count[task_state] = 1
+        # put the stats in the cache
+        self.stats_cache[framework_id] = state_count
+        return self._lookup(state_count, state)
+
+    def _lookup(self, state_count, state):
+        if state in state_count.keys():
+            return state_count[state]
+        else:
+            return 0
+
+
+def resource_stats(resource, allocated, used):
```

Comment:
This should probably be part of a new object -> "Framework" that does the calculation on these, just like Task does.